lib/logstash/inputs/http/message_handler.rb

```diff
@@ -0,0 +1,62 @@
+# encoding: utf-8
+require "logstash-input-http_jars"
+java_import org.logstash.plugins.inputs.http.MessageHandler
+java_import "io.netty.handler.codec.http.DefaultFullHttpResponse"
+java_import "io.netty.handler.codec.http.HttpHeaderNames"
+java_import "io.netty.handler.codec.http.HttpVersion"
+java_import "io.netty.handler.codec.http.HttpResponseStatus"
+java_import "io.netty.buffer.Unpooled"
+java_import "io.netty.util.CharsetUtil"
+
+module LogStash module Inputs class Http
+  class MessageHandler
+    include org.logstash.plugins.inputs.http.IMessageHandler
+
+    attr_reader :input
+
+    def initialize(input, default_codec, additional_codecs)
+      @input = input
+      @default_codec = default_codec
```

Comment:
I think we need to clone this default_codec

Reply:
it's done [here](https://github.com/logstash-plugins/logstash-input-http/pull/73/files/117a9e91fe421a9a38d675b5cc7e1bbcecfdac89#diff-67368f447845a8ce55c9184b435bdbc9R47)